### ISSUE TITLE
PdfDocument.Open throws PdfDocumentFormatException for certain PDFs

### ISSUE THREAD

**Issue by hubbardgary** [UglyToad/PdfPig#320]
Hello

I've noticed this exception when calling `PdfDocument.Open` with certain PDF files. The affected files were exported from [Lucidchart](https://www.lucidchart.com/).

Here is a sample file that results in this error:
[lucidchart-export.pdf](https://github.com/UglyToad/PdfPig/files/6404865/lucidchart-export.pdf)

I've also tried this with the `0.1.5-alpha001` release which gives the same result.

Stack trace:
```
UglyToad.PdfPig.Core.PdfDocumentFormatException: Could not find the object number 29 0 with type DictionaryToken.
   at UglyToad.PdfPig.Parser.Parts.DirectObjectFinder.Get[T](IndirectReference reference, IPdfTokenScanner scanner)
   at UglyToad.PdfPig.Parser.DocumentInformationFactory.Create(IPdfTokenScanner pdfTokenScanner, TrailerDictionary trailer)
   at UglyToad.PdfPig.Parser.PdfDocumentFactory.OpenDocument(IInputBytes inputBytes, ISeekableTokenScanner scanner, ILog log, Boolean isLenientParsing, IReadOnlyList`1 passwords, Boolean clipPaths)
   at UglyToad.PdfPig.Parser.PdfDocumentFactory.Open(IInputBytes inputBytes, ParsingOptions options)
   at UglyToad.PdfPig.Parser.PdfDocumentFactory.Open(Byte[] fileBytes, ParsingOptions options)
   at UglyToad.PdfPig.Parser.PdfDocumentFactory.Open(String filename, ParsingOptions options)
   at UglyToad.PdfPig.PdfDocument.Open(String filePath, ParsingOptions options)
   at PdfTestApp.Program.Main(String[] args) in C:\PdfTestApp\PdfTestApp\Program.cs:line 12
```

Any assistance gratefully received - thanks for all your great work on this project.

**Comment by hubbardgary** [comment]
Thank you @InusualZ, with your changes I am now able to load the files, though as you noted without any document information.

Can I assume that this fix will be included in the next release?

**Comment by InusualZ** [comment]
@EliotJones For reading the metadata stream, we would need to parse the XML in it. It would be OK to use C# Native XML API?
As of right now I have a patch ready, that allow you to read the document. But ignore the metadata stream.

What would be the best course of action here?

**Comment by EliotJones** [comment]
@InusualZ would that require a NuGet package on some runtimes/frameworks or is it just part of the core framework? I think we have some XML processing logic already here: https://github.com/UglyToad/PdfPig/blob/master/src/UglyToad.PdfPig/Content/XmpMetadata.cs

**Comment by InusualZ** [comment]
Actually I'm confuse now. From the sample document, I'm getting this in the metadata stream
```xml
{<?xpacket begin="" id="W5M0MpCehiHzreSzNTczkc9d"?>
<x:xmpmeta xmlns:x="adobe:ns:meta/" x:xmptk="Lucidchart">
  <rdf:RDF xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#">
    <rdf:Description rdf:about="" xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:xmp="http://ns.adobe.com/xap/1.0/" xmlns:xmpMM="http://ns.adobe.com/xap/1.0/mm/">
      <dc:format>application/pdf</dc:format>
      <dc:title>
        <rdf:Alt>
          <rdf:li xml:lang="x-default">Example diagram</rdf:li>
        </rdf:Alt>
      </dc:title>
      <dc:creator>
        <rdf:Bag>
          <rdf:li xml:lang="x-default">Name Surname</rdf:li>
        </rdf:Bag>
      </dc:creator>
      <dc:description>
        <rdf:Alt>
          <rdf:li xml:lang="x-default">Lucidchart</rdf:li>
        </rdf:Alt>
      </dc:description>
      <xmp:CreateDate>2021-04-30T10:33:58UTC</xmp:CreateDate>
      <xmp:ModifyDate>2021-04-30T10:33:58UTC</xmp:ModifyDate>
      <xmp:MetadataDate>2021-04-30T10:33:58UTC</xmp:MetadataDate>
      <xmpMM:DocumentID>uuid:64195f49-19bb-4f43-bc91-a4a7f4d9910f</xmpMM:DocumentID>
      <xmpMM:InstanceID>uuid:64195f49-19bb-4f43-bc91-a4a7f4d9910f</xmpMM:InstanceID>
    </rdf:Description>
  </rdf:RDF>
</x:xmpmeta>
<?xpacket end="w"?>}
```

I'm I suppose to deserialize into the document information struct? @EliotJones 
(That seems to be a xmp stream)

**Comment by EliotJones** [comment]
Ok so it looks like the `/Metadata` tag in the document catalog is a drop-in alternative for the document information dictionary. I'd say since we already expose access to this metadata we don't need to do the work of parsing it into the information struct ourselves. Since to do it properly we'd need some partial implementation of the XMP spec it's a lot of work, since consumers can access the data they can bring-their-own XMP parsing approach. I think we're ok just to have the fix where we handle the bug and return an empty info dictionary.

**Comment by InusualZ** [comment]
Fixed by #322

**Comment by hubbardgary** [comment]
Thank you so much @InusualZ amd @EliotJones.
